### ISSUE TITLE
explosion event cancellation is broke

### ISSUE THREAD

**Issue by tastydev** [altmp/altv-issues#1601]
### Description of the problem

As soon as you register the explosion event the explosion sync of throwables will break. It doesn't matter if and what you return.
see steps below

### Reproduction steps

1. Use a clean gamemode
2. connect to it with 2 players
3. throw a grenade Result: Explosion synced
4. only register this serverside event (i.e with runtime editor): 
```js

alt.on('explosion', (source, type, pos, fx, target) => {
  console.log('explosion', source.name, type, pos, fx);
  //return true; doesn't work too
});
```
5. retry step 2 and step 3 -> sync broke

### Expected behaviour

```js
//still sync the explosion
alt.on('explosion', (source, type, pos, fx, target) => {
  console.log('explosion', source.name, type, pos, fx);
});
```

```js
//still sync the explosion
alt.on('explosion', (source, type, pos, fx, target) => {
  console.log('explosion', source.name, type, pos, fx);
  return true;
});
```

```js
//don't sync the explosion
alt.on('explosion', (source, type, pos, fx, target) => {
  console.log('explosion', source.name, type, pos, fx);
  return false;
});
```

### Additional context

used linux server binaries
client version: 12.10 release
server version: 12.10 release

maybe it affects other cancelable events too

### Operating system

Ubuntu 20.04

### Version

12.10 release

**Comment by FabianTerhorst** [comment]
Can you test if its a js or core issue?

**Comment by tastydev** [comment]
> Can you test if its a js or core issue?

i tested it with c#-module too now and it's the same...
so it's core related

**Comment by tastydev** [comment]
more info: 
POV thrower: everything is working as intended
POV target: sees grenade/molotov flying, sees no explosion but gets dmg... for molotovs the fire on the ground is synced but the explosion of the molotov isn't synced, too

**Comment by Shino58** [comment]
Hi , i've the same probleme with an only if into the event without any return
as tasty say , the explosion work on thrower but not synced on target ( tested with vehicle explosion and rpg)

**Comment by unnick94** [comment]
Would fix it as soon as possible, it's impossible to fight cheaters without it

**Comment by FabianTerhorst** [comment]
Fixed in 12.22

**Comment by xxshady** [comment]
> more info: POV thrower: everything is working as intended POV target: sees grenade/molotov flying, for grenades sees no explosion but gets dmg... for molotovs the fire on the ground is synced but the explosion of the molotov isn't synced, too

@tastydev for damage you also need to cancel [weaponDamage](https://xxshady.github.io/custom-altv-types/interfaces/_alt_server_.iserverevent.html#weapondamage) event
